sihl-web/src/middleware_csrf.ml

```diff
@@ -142,20 +143,23 @@ struct
               match not_allowed_handler with
               | None ->
                 (* Give 403 if provided secret doesn't match session secret *)
-                Sihl_type.Http_response.(of_plain_text ~status:`Forbidden "")
-                |> Lwt.return
+                Opium.Response.(of_plain_text ~status:`Forbidden "") |> Lwt.return
               | Some handler -> Lwt.return @@ handler req)
             else
               (* Provided secret matches and is valid => Invalidate it so it can't be
                  reused *)
               let* () = TokenService.invalidate ps in
+              (* To allow fetching a new valid token from the context, generate a new one *)
+              let* secret = create_secret session in
```

Comment:
These three lines are the core change of the fix FYI

Reply:
Thanks for the hint!